cirq-core/cirq/circuits/circuit.py

```diff
@@ -1362,8 +1364,10 @@ def to_qasm(
             qubit_order: Determines how qubits are ordered in the QASM
                 register.
         """
+        output_precision = args.precision if args else precision
```

Comment:
This overwrites the `precision` argument.  Perhaps we could add a new `version` argument here instead and change the `_qasm_` function to use `_to_qasm_output` instead of `to_qasm`.

This patch - [circuit-to_qasm-with-version.patch.txt](https://github.com/user-attachments/files/17783034/circuit-to_qasm-with-version.patch.txt) - does so and seems to pass the tests.

Reply:
Patch applied.